scrapi/harvesters/dataone.py

```diff
@@ -183,4 +183,8 @@ def get_records(self, start_date, end_date):
             docs = etree.XML(data.content).xpath('//doc')
             for doc in docs:
                 yield doc
-            n += 1000
+
+            if page_limit and int(page_limit) == n / 1000:
```

Comment:
This may have different behavior in python3 (division operator semantics changed slightly, IIRC)